src/query/fuzzy_query.rs

```diff
@@ -100,10 +101,16 @@ impl FuzzyTermQuery {
     }
 
     fn specialized_weight(&self) -> Result<AutomatonWeight<DFA>> {
-        let automaton = LEV_BUILDER.get(&(self.distance, false))
-            .unwrap() // TODO return an error
-            .build_dfa(self.term.text());
-        Ok(AutomatonWeight::new(self.term.field(), automaton))
+        // LEV_BUILDER is a HashMap, whose `get` method returns an Option
+        match LEV_BUILDER.get(&(self.distance, false)) {
+            // Unwrap the option and build the AutomatonWeight
+            Some(automaton_builder) => {
+                let automaton = automaton_builder.build_dfa(self.term.text());
+                Ok(AutomatonWeight::new(self.term.field(), automaton))
+            }
+            // Err - maybe there is a better Error type
```

Comment:
This is about being poisoned. It's more like an invalid argument. The LevenshteinBuilder only accepts distances of 0,1, or 2.